test/contrib/s3_test.py

```diff
@@ -107,6 +108,7 @@ def test_read_no_file_sse(self):
         t = self.create_target(encrypt_key=True)
         self.assertRaises(FileNotFoundException, t.open)
 
+    @unittest.skipIf(tuple(sys.version_info) >= (3, 12), "boto is not supported on Python 3.12+")
```

Comment:
I think the better thing to do here would be to make the s3 module compatible with boto3

Reply:
Can the s3 module be improved in another PR? Ubuntu 24 came out with Python 3.12 as default, and the lack of support in Luigi will block updates for many users (although fortunately, the changes only affect tests, so it's not critical).

Reply:
Given the need for 3.12 Luigi support, i'm ok getting this PR out the door.

@ukky17 , if you can resolve the merge conflict, i can merge if tests still pass.

Reply:
@dlstadther I have resolved the conflict. Please review again and merge if it looks good to you.